src/simplewallet/simplewallet.cpp

```diff
@@ -1660,6 +1662,60 @@ bool simple_wallet::handle_command_line(const boost::program_options::variables_
 
   return true;
 }
+//----------------------------------------------------------------------------------------------------
+bool simple_wallet::prompt_for_new_node()
+{
+  bool connect = false;
+  std::string node_url;
+
+  std::string confirm = input_line(tr("Do you want to connect to a node?  (Y/Yes/N/No): "));
```

Comment:
Why not directly take the daemon URL from the command argument?

Also I'm not so sure if a confirmation is really necessary.

Reply:
I think this is because it can be called either with "set_node" (in which case it should take an argument as you say) and if connection fails (in which case it allows the user to give a new one). That should be how functions are separated: one with an arument which does not prompt, and one which asks "do you want", parses, then calls the first one with the input.

Reply:
> if connection fails (in which case it allows the user to give a new one

Do you mean that this prompt is supposed to be called automatically when a connection gets lost? But I don't see the function `prompt_for_new_node` being called from anywhere other than `set_node`. Am I missing something?

Reply:
I think it used to. Or maybe I just misremembered :)